drop TABLE gnomad;

CREATE TABLE gnomad(
	"chromosome" INT,
	"position" INT,
	"allele_ref" VARCHAR(255),
	"allele_alt" VARCHAR(255),
	"qual" INT,
	"filters" VARCHAR(20),
	"vartype" VARCHAR(255),
	"gene" VARCHAR(255),
	"transcript_consequence" VARCHAR(255),
	"protein_consequence" VARCHAR(255),
	"codon_num" INT,
	"ref_aa" VARCHAR(255),
	"alt_aa" VARCHAR(255),
	"FS" REAL,
	"MQRankSum" REAL,
	"InbreedingCoeff" REAL,
	"ReadPosRankSum" REAL,
	"VQSLOD" REAL,
	"QD" REAL,
	"DP" REAL,
	"BaseQRankSum" REAL,
	"MQ" REAL,
	"ClippingRankSum" REAL,
	"rf_tp_probability" REAL,
	"pab_max" REAL,
	"AC" INT,
	"AN" INT,
	"non_neuro_AC" INT,
	"non_neuro_AN" INT,
	"non_topmed_AC" INT,
	"non_topmed_AN" INT,
	"controls_AC" INT,
	"controls_AN" INT,
	"source" VARCHAR(7)
);